rust-connectors/sources/mqtt/src/main.rs

```diff
@@ -84,11 +97,40 @@ fn main() -> Result<(), MqttConnectorError> {
             %mqtt_topic,
             %client_id
         );
+        let port = if let Some(port) = opts.mqtt_port {
+            port
+        } else {
+            if opts.mqtt_tls {
+                8883
+            } else {
+                1883
+            }
+        };
 
         let timeout = std::time::Duration::from_secs(mqtt_timeout_seconds);
 
-        let mut mqttoptions = MqttOptions::new(client_id, mqtt_url, 1883);
+        let mut mqttoptions = MqttOptions::new(client_id, mqtt_url, port);
+        if let (Some(username), Some(password)) = (opts.mqtt_username, opts.mqtt_password) {
+            mqttoptions.set_credentials(username, password);
+        }
         mqttoptions.set_keep_alive(timeout);
+        if opts.mqtt_tls {
+            // Use rustls-native-certs to load root certificates from the operating system.
```

Comment:
we probably need something for cert management overall perhaps coordinated across connectors. 

also probably we should perhaps also do tests to ensure we validate cert negative/positive.